x/market/keeper/swap.go

```diff
@@ -127,13 +127,19 @@ func (k Keeper) ComputeSwap(ctx sdk.Context, offerCoin sdk.Coin, askDenom string
 		spread = minSpread
 	}
 
+	var errSup = k.ValidateSupplyMaximum(ctx, sdk.NewDecCoin(retDecCoin.Denom, retDecCoin.Amount.TruncateInt()))
```

Comment:
In my opinion it would be much cleaner if you:

- Removed `ValidateSupplyMaximum()` from `ComputeSwap()` and `ComputeInternalSwap()`
- Added the max supply check in the `handleSwapRequest()` which is directly called by the msg handler

This way you would call into the validation routine only once and at a single point (namely after the following line: https://github.com/classic-terra/core/blob/main/x/market/keeper/msg_server.go#L79). Additionally, you would not have to worry about the breaking tests and changing the interface of `ComputeSwap()`

Reply:
Adjusted as directed!